### PR TITLE
rustfmt 1.3.2 (new formula)

### PR DESCRIPTION
**Pull request by zengxs**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

[rustfmt](https://github.com/rust-lang/rustfmt) is a tool for formatting Rust code according to style guidelines.

### CODE REVIEW

**Comment by lembacon** [comment]
If this only builds against the nightly build of Rust, I think we do not want to add it to core then. cc @Homebrew/maintainers

File: Formula/rustfmt.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  sha256 "d9ff195fa0f02012c0be8aa7b387e2140271809f61f3b621d7228a338670e040"
+
+  depends_on "rustup-init" => :build
+
+  depends_on "rust"
+
+  def install
+    ENV["RUSTUP_HOME"] = buildpath/"rustup"
+    ENV["CARGO_HOME"] = buildpath/"cargo"
+    system "rustup-init", "--default-toolchain", "nightly", "-y"
```
Comment (sjackman) [inline Formula/rustfmt.rb]:
```suggestion
```
Please remove this line.

File: Formula/rustfmt.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    system buildpath/"cargo/bin/cargo", "install", "--root", prefix, "--path", "."
```
Comment (sjackman) [inline Formula/rustfmt.rb]:
```suggestion
    system "cargo", "install", "--root", prefix, "--path", "."
```

File: Formula/rustfmt.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+class Rustfmt < Formula
+  desc "Rust code formatter"
+  homepage "https://github.com/rust-lang/rustfmt"
+  url "https://github.com/rust-lang/rustfmt/archive/v1.3.2.tar.gz"
+  sha256 "d9ff195fa0f02012c0be8aa7b387e2140271809f61f3b621d7228a338670e040"
+
+  depends_on "rustup-init" => :build
+
+  depends_on "rust"
```
Comment (sjackman) [inline Formula/rustfmt.rb]:
```suggestion
  depends_on "rust" => build
```

File: Formula/rustfmt.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+class Rustfmt < Formula
+  desc "Rust code formatter"
+  homepage "https://github.com/rust-lang/rustfmt"
+  url "https://github.com/rust-lang/rustfmt/archive/v1.3.2.tar.gz"
+  sha256 "d9ff195fa0f02012c0be8aa7b387e2140271809f61f3b621d7228a338670e040"
+
+  depends_on "rustup-init" => :build
```
Comment (sjackman) [inline Formula/rustfmt.rb]:
```suggestion
```
Please remove this line.

File: Formula/rustfmt.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+  depends_on "rust"
+
+  def install
+    ENV["RUSTUP_HOME"] = buildpath/"rustup"
```
Comment (sjackman) [inline Formula/rustfmt.rb]:
```suggestion
```
Please remove this line.

File: Formula/rustfmt.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    ENV["CARGO_HOME"] = buildpath/"cargo"
```
Comment (sjackman) [inline Formula/rustfmt.rb]:
```suggestion
```
Please remove this line.

**Review by sjackman — CHANGES_REQUESTED** [review]
Thank you for your contribution to Homebrew, @zengxs! I would also like to see `rustfmt` in Homebrew. Please the the brewed version of `rust`.

**Pull request closed without merging**